File: app/src/main/java/com/jameschamberlain/footballteamtracker/Player.java
package com.jameschamberlain.footballteamtracker;

import android.os.Parcel;
import android.os.Parcelable;

import androidx.annotation.NonNull;

import java.io.Serializable;

public class Player implements Parcelable, Comparable<Player> {

    private String name;
    private int goals;
    private int assists;

    public Player(String name) {
        this.name = name;
        this.goals = 0;
        this.assists = 0;
    }

    protected Player(Parcel in) {
        name = in.readString();
        goals = in.readInt();
        assists = in.readInt();
    }

    public static final Creator<Player> CREATOR = new Creator<Player>() {
        @Override
        public Player createFromParcel(Parcel in) {
            return new Player(in);
        }

        @Override
        public Player[] newArray(int size) {
            return new Player[size];
        }
    };

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public int getGoals() {
        return goals;
    }

    public void setGoals(int goals) {
        this.goals = goals;
    }

    public int getAssists() {
        return assists;
    }

    public void setAssists(int assists) {
        this.assists = assists;
    }

    @NonNull
    @Override
    public String toString() {
        return "Name: " + this.name + ", Goals: " + this.goals + ", Assists: " + this.assists;
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeString(name);
        dest.writeInt(goals);
        dest.writeInt(assists);
    }

    @Override
    public int compareTo(Player o) {
        return getName().compareTo(o.getName());
    }
}

File: app/src/main/java/com/jameschamberlain/footballteamtracker/data/Player.kt
package com.jameschamberlain.footballteamtracker.data

import android.os.Parcel
import android.os.Parcelable

class Player() : Parcelable, Comparable<Player> {
    lateinit var name: String
    var goals: Int = 0
    var assists: Int = 0

    constructor(parcel: Parcel) : this() {
        name = parcel.readString().toString()
        goals = parcel.readInt()
        assists = parcel.readInt()
    }


    constructor(name: String) : this() {
        this.name = name
        goals = 0
        assists = 0
    }

    override fun toString(): String {
        return "Name: $name, Goals: $goals, Assists: $assists"
    }

    override fun compareTo(other: Player): Int {
        return name.compareTo(other.name)
    }

    override fun writeToParcel(parcel: Parcel, flags: Int) {
        parcel.writeString(name)
        parcel.writeInt(goals)
        parcel.writeInt(assists)
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object CREATOR : Parcelable.Creator<Player> {
        override fun createFromParcel(parcel: Parcel): Player {
            return Player(parcel)
        }

        override fun newArray(size: Int): Array<Player?> {
            return arrayOfNulls(size)
        }
    }
}